### ISSUE TITLE
Editor adds extra <br/> every time I edit/save my document

### ISSUE THREAD

**Issue by mstermaaten** [ckeditor/ckeditor5#5589]
I run and admin panel for my blog that uses Ckeditor5 with Firebase to store my content.

Every time I open an previous created document to edit it, the editor will automatically add extra <br/> tags in the <p/> that creates the actual break in my document.
But because this happens every time i want to edit my document these breaks become huge if i don't keep removing breaks after every edit. 

I couldn't find anything that worked in the documentation that would fix this, maybe i'm missing something??

Recreate: 

1. Setup React/editor component (npm i @ckeditor/ckeditor5) with firebase
2. Write article with a break and store editor content with getData() in Firebase
3. Edit document by initializing editor with content parsed in the CKEditor.data={Firebase content}.

![Schermafdruk 2019-10-15 14 42 17](https://user-images.githubusercontent.com/49290062/66833280-d8ceba80-ef5b-11e9-9074-03ce18dd4664.png)

### ✔️ Expected result

![Schermafdruk 2019-10-15 14 40 43](https://user-images.githubusercontent.com/49290062/66832477-4aa60480-ef5a-11e9-92bb-04ff78974da6.png)

### ❌ Actual result
This is the result after Saving & opening in editor 2 times:

![Schermafdruk 2019-10-15 14 41 04](https://user-images.githubusercontent.com/49290062/66832495-5396d600-ef5a-11e9-89ea-ca51afe4e57f.png)

![Schermafdruk 2019-10-15 14 41 56](https://user-images.githubusercontent.com/49290062/66832856-0109e980-ef5b-11e9-90c2-d537729daf46.png)

## 📃 Other details

* Browser: Chrome
* OS: IOS Mac
* CKEditor version:  "@ckeditor/ckeditor5-react": "^1.1.3",
* Installed CKEditor plugins:

![Schermafdruk 2019-10-15 14 42 17](https://user-images.githubusercontent.com/49290062/66832833-f8b1ae80-ef5a-11e9-9169-33c3907914bd.png)

-     "@ckeditor/ckeditor5-adapter-ckfinder": "^11.0.5",
-     "@ckeditor/ckeditor5-alignment": "^11.2.0",
-     "@ckeditor/ckeditor5-autoformat": "^11.0.5",
-     "@ckeditor/ckeditor5-basic-styles": "^11.1.4",
-     "@ckeditor/ckeditor5-block-quote": "^11.1.3",
-     "@ckeditor/ckeditor5-ckfinder": "^11.0.5",
-     "@ckeditor/ckeditor5-editor-classic": "^12.1.4",
-     "@ckeditor/ckeditor5-core": "^12.3.0",
-     "@ckeditor/ckeditor5-dev-utils": "^12.0.0",
-     "@ckeditor/ckeditor5-dev-webpack-plugin": "^8.0.0",
-     "@ckeditor/ckeditor5-easy-image": "^11.0.5",
-     "@ckeditor/ckeditor5-essentials": "^11.0.5",
-     "@ckeditor/ckeditor5-font": "^11.2.2",
-     "@ckeditor/ckeditor5-heading": "^11.0.5",
-     "@ckeditor/ckeditor5-highlight": "^11.0.5",
-     "@ckeditor/ckeditor5-horizontal-rule": "0.0.1",
-     "@ckeditor/ckeditor5-image": "^14.0.0",
-     "@ckeditor/ckeditor5-indent": "^10.1.0",
-     "@ckeditor/ckeditor5-link": "^11.1.2",
-     "@ckeditor/ckeditor5-list": "^12.1.0",
-     "@ckeditor/ckeditor5-media-embed": "^11.1.4",
-     "@ckeditor/ckeditor5-paragraph": "^11.0.5",
-     "@ckeditor/ckeditor5-paste-from-office": "^11.1.0",
-     "@ckeditor/ckeditor5-react": "^1.1.3",
-     "@ckeditor/ckeditor5-table": "^14.0.0",
-     "@ckeditor/ckeditor5-theme-lark": "^14.2.0",
-     "@ckeditor/ckeditor5-upload": "^12.0.0",

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.****

**Comment by mstermaaten** [comment]
I think it has something to do with the `data-cke-filler="true"` it adds to the `<p><br/></p>`.

**Comment by andreyalikberov** [comment]
I've also run into this issue. Does anyone know when it is going to be resolved? @CKEditorCLA @CKEditorBot

**Comment by Reinmar** [comment]
Hi! I tried to reproduce this but all works fine for me. Could you create a minified test scenario on which this would be reproducible?

**Comment by mstermaaten** [comment]
[https://github.com/mstermaaten/test-editor](https://github.com/mstermaaten/test-editor)

This is a basic version of my product with fake credentials but using the same configs and packages.

[https://objective-blackwell-ced0b4.netlify.com/](https://objective-blackwell-ced0b4.netlify.com/)
username: test@editor.com
pw: testing

**Comment by mstermaaten** [comment]
I'm having more issues now, especially the error code: 
_"view-position-before-root: You can not make position before root"._

<img width="612" alt="Schermafbeelding 2019-11-01 om 16 07 06" src="https://user-images.githubusercontent.com/49290062/68034336-c5448300-fcc1-11e9-9565-ef94069fd7e6.png">

I really have the feeling that I might be missing something that i'm not able to see based on documentation. Maybe something is missing in my web-pack.config?
Or that I'm trying to add to many plugins in my editor build, could this be?
Maybe because i'm using firebase to store my data?

If anybody could have a look at my code that would be very appreciate:
https://github.com/mstermaaten/test-editor

**Comment by Reinmar** [comment]
I tried to run your app but it fails on missing Firebird configuration (or something similar). Meaning – it's definitely not a minimal example of the issue. By minimal I mean something that does not involve anything else than the editor.

Anyway, I noticed that `<br data-cke-filler=true>` is saved in your database. I confirmed that the editor returns this element instead of a correct filler when MathType plugins are enabled. This is incorrect, because this filler must be used only in the editing pipeline, not in  the data. Meaning – the MathType plugin is making the editor return something completely incorrect.
